### PR TITLE
[delta xds]: remove cached nonces on reconnection

### PR DESCRIPTION
**Pull request by stevenzzzz**
Commit Message: See https://github.com/envoyproxy/envoy/issues/15767, remove cached nonces on reconnection. 
Additional Description:
Risk Level: LOW
Testing: unit test
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Fixes #Issue] #15767

### CODE REVIEW

**Comment by stevenzzzz** [comment]
@yanavlasov @htuch PTAL

File: source/common/common/interval_value.h [hunk 1]
```diff
@@ -25,10 +25,11 @@ template <typename T, typename Interval> class ClosedIntervalValue {
...
   ClosedIntervalValue invert() const {
-    return ClosedIntervalValue(value_ == Interval::max_value ? Interval::min_value
-                               : value_ == Interval::min_value
-                                   ? Interval::max_value
-                                   : Interval::max_value - (value_ - Interval::min_value));
+    return ClosedIntervalValue(value_ == Interval::max_value
+                                   ? Interval::min_value
+                                   : value_ == Interval::min_value
+                                         ? Interval::max_value
+                                         : Interval::max_value - (value_ - Interval::min_value));
```
Comment (htuch) [inline source/common/common/interval_value.h:32]: @stevenzzzz format is failing, and it seems this is a gratuitous change, so maybe take a look into that.

File: test/common/network/apple_dns_impl_test.cc [hunk 2]
```diff
@@ -350,12 +350,12 @@ TEST_F(AppleDnsImplFakeApiTest, SynchronousErrorInGetAddrInfo) {
   // The Query's sd ref will be deallocated.
   EXPECT_CALL(dns_service_, dnsServiceRefDeallocate(_));
 
-  EXPECT_EQ(nullptr,
-            resolver_->resolve("foo.com", Network::DnsLookupFamily::Auto,
-                               [](DnsResolver::ResolutionStatus, std::list<DnsResponse>&&) -> void {
-                                 // This callback should never be executed.
-                                 FAIL();
-                               }));
+  EXPECT_EQ(nullptr, resolver_->resolve(
```
Comment (htuch) [inline test/common/network/apple_dns_impl_test.cc:353]: Yeah, please fix formating :)

File: test/common/config/new_grpc_mux_impl_test.cc [hunk 3]
```diff
@@ -123,6 +130,52 @@ TEST_F(NewGrpcMuxImplTest, DynamicContextParameters) {
   expectSendMessage("foo", {}, {"x", "y"});
 }
 
+// Validate behavior when dynamic context parameters are updated.
```
Comment (htuch) [inline test/common/config/new_grpc_mux_impl_test.cc:133]: This doesn't seem the right description.

File: test/common/config/new_grpc_mux_impl_test.cc [hunk 4]
```diff
@@ -123,6 +130,52 @@ TEST_F(NewGrpcMuxImplTest, DynamicContextParameters) {
...
+          testing::DoAll(SaveArg<0>(&grpc_stream_retry_timer_cb), Return(grpc_stream_retry_timer)))
+      // Happens when add first watch.
+      .WillRepeatedly(Return(ttl_mgr_timer));
+  setup();
+  InSequence s;
+  const std::string& type_url = Config::TypeUrl::get().ClusterLoadAssignment;
+  auto foo_sub = grpc_mux_->addWatch(type_url, {"x", "y"}, callbacks_, resource_decoder_, {});
+  EXPECT_CALL(*async_client_, startRaw(_, _, _, _)).WillOnce(Return(&async_stream_));
+  expectSendMessage(type_url, {"x", "y"}, {});
+  grpc_mux_->start();
```
Comment (htuch) [inline test/common/config/new_grpc_mux_impl_test.cc:149]: Please add some more comments.

File: test/common/config/new_grpc_mux_impl_test.cc [hunk 5]
```diff
@@ -123,6 +130,52 @@ TEST_F(NewGrpcMuxImplTest, DynamicContextParameters) {
...
+  auto response = std::make_unique<envoy::service::discovery::v3::DeltaDiscoveryResponse>();
+  response->set_type_url(type_url);
+  response->set_system_version_info("3000");
+  response->set_nonce("111");
+  envoy::config::endpoint::v3::ClusterLoadAssignment cla;
+  cla.set_cluster_name("xxx");
+  auto res = response->add_resources();
```
Comment (htuch) [inline test/common/config/new_grpc_mux_impl_test.cc:156]: Prefer structuring this using helper utils or something like that.

**Review by htuch — CHANGES_REQUESTED** [review]
Makes sense to me, just needs some cleanup, thanks for the fix!
/wait

**Review by stevenzzzz — COMMENTED** [review]
thanks

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**